src/core/engine.rs

```diff
@@ -1163,9 +1186,28 @@ impl LLMEngine {
         }
         let mut receivers = Vec::new();
         for (param, messages) in params.iter().zip(message_list.iter()) {
-            let (prompt, image_idx) = self.apply_chat_template(param, messages, false);
+            // Fingerprint-based automatic session detection
+            let mut param = param.clone();
+            if param.session_id.is_none()
+                && self.econfig.flash_context.unwrap_or(false)
```

Comment:
@guoqingbao - i would think fiash_context would _help_ with this... what does it break?